src/imp/security_framework.rs

```diff
@@ -385,6 +411,25 @@ impl<S: io::Read + io::Write> TlsStream<S> {
         Ok(trust.certificate_at_index(0).map(Certificate))
     }
 
+    pub fn certificate_chain(&mut self) -> Result<ChainIterator<S>, Error> {
+        let trust = match self.stream.context().peer_trust2()? {
+            Some(trust) => trust,
+            None => {
+                return Ok(ChainIterator {
```

Comment:
This can be restructured a bit to avoid having to have a separate early return:

```rust
let trust = match self.stream.context().peer_trust2()? {
    Some(trust) => {
        trust.evaluate()?;
        Some(trust)
    }
    None => None
};
...
```